internal/controller/api/v1alpha1/host_types.go

```diff
@@ -17,25 +17,117 @@ limitations under the License.
 package v1alpha1
 
 import (
+	"github.com/openshift/assisted-service/models"
+	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
-// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+type HostMemory struct {
+	PhysicalBytes int64 `json:"physicalBytes,omitempty"`
+	UsableBytes   int64 `json:"usableBytes,omitempty"`
+}
+
+type HostCPU struct {
+	Count        int64    `json:"count,omitempty"`
+	Frequency    int64    `json:"frequency,omitempty"`
+	Flags        []string `json:"flags"`
+	ModelName    string   `json:"modelName,omitempty"`
+	Architecture string   `json:"architecture,omitempty"`
+}
+
+type HostInterface struct {
+	IPV6Addresses []string `json:"ipv6Addresses"`
+	Vendor        string   `json:"vendor,omitempty"`
+	Name          string   `json:"name,omitempty"`
+	HasCarrier    bool     `json:"hasCarrier,omitempty"`
+	Product       string   `json:"product,omitempty"`
+	Mtu           int64    `json:"mtu,omitempty"`
+	IPV4Addresses []string `json:"ipv4Addresses"`
+	Biosdevname   string   `json:"biosDevname,omitempty"`
+	ClientID      string   `json:"clientId,omitempty"`
+	MacAddress    string   `json:"macAddress,omitempty"`
+	Flags         []string `json:"flags"`
+	SpeedMbps     int64    `json:"speedMbps,omitempty"`
+}
+
+type HostInstallationEligibility struct {
+	Eligible           bool     `json:"eligible,omitempty"`
+	NotEligibleReasons []string `json:"not_eligibleReasons"`
+}
+
+type HostIoPerf struct {
+	// 99th percentile of fsync duration in milliseconds
+	SyncDuration int64 `json:"syncDuration,omitempty"`
+}
+
+type HostDisk struct {
+	DriveType               string                      `json:"driveType,omitempty"`
+	Vendor                  string                      `json:"vendor,omitempty"`
+	Name                    string                      `json:"name,omitempty"`
+	Path                    string                      `json:"path,omitempty"`
+	Hctl                    string                      `json:"hctl,omitempty"`
+	ByPath                  string                      `json:"byPath,omitempty"`
+	Model                   string                      `json:"model,omitempty"`
+	Wwn                     string                      `json:"wwn,omitempty"`
+	Serial                  string                      `json:"serial,omitempty"`
+	SizeBytes               int64                       `json:"sizeBytes,omitempty"`
+	Bootable                bool                        `json:"bootable,omitempty"`
+	Smart                   string                      `json:"smart,omitempty"`
+	InstallationEligibility HostInstallationEligibility `json:"installationEligibility,omitempty"`
+	IoPerf                  HostIoPerf                  `json:"ioPerf,omitempty"`
```

Comment:
Made a related comment about `InstallationEligibility` and `IoPerf` at [enhancements/pull/574](
https://github.com/openshift/enhancements/pull/574/files#r551009802)